dashboard/app/models/pd/workshop.rb

```diff
@@ -580,6 +580,34 @@ def attending_teachers
     sessions.flat_map(&:attendances).flat_map(&:teacher).uniq
   end
 
+  # Get all teachers who have attended all sessions of this workshop.
+  def teachers_attending_all_sessions(cdo_scholarship=false)
+    teachers_attending = sessions.flat_map(&:attendances).flat_map(&:teacher)
+
+    # Filter attendances to only scholarship teachers
+    if cdo_scholarship
+      teachers_attending.select! do |teacher|
+        Pd::ScholarshipInfo.exists?(
```

Comment:
I don't think this check is sufficient.  We have `Pd::ScholarshipInfo` rows where `scholarship_status` is `'no'`, which doesn't seem to match what you're querying for here.  In fact, I think all applications get a `ScholarshipInfo` now?

Reply:
Second concern here - we're nesting a second ActiveRecord query inside a loop, so we'll run the `exists?` once for every teacher in the workshop.  It would be better to add a `join` to the `teachers_attending` relation so we're not adding a query... but that could be follow-up work.  Let's make sure this is working as designed first.

Reply:
You're totally right re: check not being sufficient, will update. Apologies for the oversight.

Reply:
Updated with appropriate check, and changed to be a single ActiveRecord query.